### PR TITLE
Let logspace support bfloat16 on both CPU and CUDA

### PR DESCRIPTION
**Pull request by xuhdev**

### CODE REVIEW

File: test/test_tensor_creation_ops.py [hunk 1]
```diff
@@ -732,7 +732,7 @@ def test_linspace_vs_numpy(self, device, dtype):
 
     @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
     @precisionOverride({torch.float: 1e-6, torch.double: 1e-10})
-    @dtypes(torch.float, torch.double)
+    @dtypes(*torch.testing.get_all_fp_dtypes(include_half=False, include_bfloat16=False))
```
Comment (mruberry) [inline test/test_tensor_creation_ops.py:735]: This is OK but you're not testing bfloat16 in this PR. You may need to add a separate, "easier" tests that compares bfloat16 vs. NumPy's logspace in fp32.

Comment (xuhdev) [reply test/test_tensor_creation_ops.py:735]: Similar to `linspace`, I think this test simply test its consistency with NumPy. The modification of the decorator of `test_logspace` below adds bfloat16 into the test.

Comment (mruberry) [reply test/test_tensor_creation_ops.py:735]: OK; that seems reasonable.

**Review by mruberry — APPROVED** [review]
Cool!

**Pull request closed without merging**